components/organism/nestedCheckboxes/src/index.js

```diff
@@ -0,0 +1,120 @@
+import React, {useState} from 'react'
+import PropTypes from 'prop-types'
+
+import MoleculeCheckboxField from '@s-ui/react-molecule-checkbox-field'
+
+const baseClass = 'sui-OrganismNestedCheckboxes'
+
+const noop = () => {}
+
+const OrganismNestedCheckboxes = ({
+  id,
+  labelParent = '',
+  onChangeParent = noop,
+  items = [],
+  join = false,
+  onChangeItem = noop,
+  fullCheckedStyledIcon: FullCheckedStyledIcon,
+  halfCheckedStyledIcon: HalfCheckedStyledIcon,
+  defaultItemsVisibility = true,
+  showItemsIcon: ShowItemsIcon,
+  hideItemsIcon: HideItemsIcon
+}) => {
+  const [showItems, setShowItems] = useState(defaultItemsVisibility) //eslint-disable-line
+  const showToggleIcons = ShowItemsIcon !== null
+
+  const parentIsFullChecked = items.every(({checked}) => checked === true)
+  const parentIsHalfChecked =
+    !parentIsFullChecked && items.some(({checked}) => checked === true)
+
+  const parentIsFullOrHalfChecked = parentIsFullChecked || parentIsHalfChecked
+
+  const parentStyledIcon = parentIsHalfChecked
+    ? HalfCheckedStyledIcon
+    : FullCheckedStyledIcon
+
+  return (
+    <ul className={baseClass}>
+      <li>
+        <MoleculeCheckboxField
```

Comment:
I consider it as an improvement that this component could have been another one like `NestedToggleCheckboxField` that had the new toggle icon props but inside `MoleculeCheckboxField` render.

```jsx
function NestedToggleCheckboxField(props) {
  render <MoleculeCheckboxField />
}
```

In this way we do not add props from a specific case to the props API of the existing `MoleculeCheckboxField` component

Reply:
I think it is not a good practice for SUI Components. With this, we can have a lot of components.